src/utilities/auth.tsx

```diff
@@ -1,10 +1,5 @@
 import { RegisterOptions } from 'react-hook-form';
 
-export interface ErrorMessageType {
-  msg: string;
-  value?: string;
-  param?: string;
-}
 interface ValidationRuleProp {
   [x: string]: RegisterOptions;
 }
```

Comment:
could you explain how this `[x: string]` works Janice? I've never seen it

Reply:
yup, so since there's a lot of fields here and they're all pointing to `RegisterOptions`, I decided to instead just do a dynamic key for the prop object (https://stackoverflow.com/questions/39256682/how-to-define-an-interface-for-objects-with-dynamic-keys). It'll match each of the `key` under that prop as long as the `key` is a string